### PR TITLE
Fixing a node's touch event to act on another node

### PR DESCRIPTION
**Pull request by qiuguohua**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/scene-graph/node-event-processor.ts [hunk 1]
```diff
@@ -162,6 +162,17 @@ export class NodeEventProcessor {
                 child._eventProcessor.setEnabled(value, true);
             }
         }
+        // If a touch event is being received for processing and a disabled node is set.
+        // The touch cancel event needs to be triggered manually. 
+        // Otherwise, it may cause the touch operation of a different node to affect the previous node.
```
Comment (minggo) [inline cocos/scene-graph/node-event-processor.ts]: - Please polish the comments.
- please fix eslint error

File: cocos/scene-graph/node-event-processor.ts [hunk 2]
```diff
@@ -162,6 +162,17 @@ export class NodeEventProcessor {
...
+        if (this._dispatchingTouch) {
+            // Dispatch touch cancel event when node is destroyed.
+            const cancelEvent = new EventTouch([this._dispatchingTouch], true, InputEventType.TOUCH_CANCEL);
+            cancelEvent.touch = this._dispatchingTouch;
+            this.dispatchEvent(cancelEvent);
+            this.claimedTouchIdList.splice(0, this.claimedTouchIdList.length);
```
Comment (PPpro) [inline cocos/scene-graph/node-event-processor.ts]: we use `this.claimedTouchIdList.length = 0;`  to clear the array

Comment (qiuguohua) [reply cocos/scene-graph/node-event-processor.ts]: OK

**Review by PPpro — APPROVED** [review]

File: cocos/scene-graph/node-event-processor.ts [hunk 3]
```diff
@@ -162,6 +162,19 @@ export class NodeEventProcessor {
                 child._eventProcessor.setEnabled(value, true);
             }
         }
+        // When a node is dispatching touch events and the node is set to disabled,
+        // The dispatching events function will hang until the node is enabled.
```
Comment (minggo) [inline cocos/scene-graph/node-event-processor.ts]: The -> the

File: cocos/scene-graph/node-event-processor.ts [hunk 4]
```diff
@@ -162,6 +162,19 @@ export class NodeEventProcessor {
...
+        // If the node is re-enabled, any touch events will be handled by this node,
+        // even if the touch events are not in the scope of this node.This is an error.
```
Comment (minggo) [inline cocos/scene-graph/node-event-processor.ts]: A space is needed after `node.`.

File: cocos/scene-graph/node-event-processor.ts [hunk 5]
```diff
@@ -162,6 +162,19 @@ export class NodeEventProcessor {
...
+        // So, Sending a cancel event when the node is set to disable.
```
Comment (minggo) [inline cocos/scene-graph/node-event-processor.ts]: - Sending -> sending
- disable -> disabled

**Review by minggo — APPROVED** [review]

**Pull request merged**